src/neo/SmartContract/Helper.cs

```diff
@@ -130,6 +130,49 @@ public static UInt160 ToScriptHash(this ReadOnlySpan<byte> script)
             return new UInt160(Crypto.Hash160(script));
         }
 
+        public static bool TryCreateVerifyEngine(this IVerifiable verifiable, StoreView snapshot, UInt160 hash, int witnessIndex, long gas, out ApplicationEngine engine)
```

Comment:
What about return an `IEnumerable<ApplicationEngine>`?

Reply:
For each witness?

Reply:
Yes.

Reply:
Please, take a look

Reply:
If we return a IEnumerable, we will need to `Execute` in the same method...

Reply:
You want to execute them in different RPC requests?

Reply:
The goal it's access to the request stack, maybe we should execute and Peek instead of pop? 🤔

Reply:
Also it will create a `ExecutionEngine` when doesn't meet the `stateDependant` filter. I think that it's better without the `IEnumerable` what do you think?

Reply:
Ping, There are 3 PRs blocked by this pr.

Reply:
I reverted the change, we should wait for @erikzhang 's review